Linux Installation Documentation Fixes and Changes

coreypjt opened the pull request:
Heyo! I'm making this pull request as I just received my elbow upgrade before my enhanced core set arrived, and luckily it allowed me to get a head start on making sure everything was set up so that I can make sure everything goes smoothly when the rest of the trackers eventually arrive.
However, during my setting up, I was following the documentation, and I discovered that a lot of the stuff that was required through set up was not documented properly. 

Here were my main problems:
- The common desync issue that happens when SlimeVR is on BEFORE SteamVR is initialised was never in the documentation, and I was only able to fix this issue by going to the Discord server and looking up what was going on
- The documentation has zero mentions of the Flatpak app which is linked in the slimevr.dev/download website (the website which is put on the manual that comes with the trackers), meaning I had several problems with getting the program to recognise my devices.
- When I installed the AppImage, it didn't install SlimeVR Feeder App alongside it, meaning my controllers weren't working with the application. Again, this was NEVER mentioned in the documentation, meaning I had to look it up in Discord server.
- SteamVR's Desktop View doesn't work whatsoever (I've tried it on multiple OSes, both on Wayland and X11), and in order to calibrate the position of the trackers (including the headset and controllers), I have to be in a position that doesn't really allow me to take my VR headset off, mainly because it was made for an in-headset view. I eventually downloaded an external program called WlxOverlay-S, which took me a long time to find. I personally think that this should be in the documentation, because, unfortunately, a quick Google search to try and fix these problems doesn't really lead to much.

Even though I did eventually figure this all out, it would still help for it to be on the documentation for people who aren't too tech-y. _Although, people who aren't tech-y should probably stay away from Linux..._

The pull request I've created _should_ fix all of the problems I mentioned earlier, and some additional things along the way. I've written most of what I've done in the commits.

I am completely new to this sort of thing, so let me know if I need to do something on my end at all. 

Thank you!

coreypjt committed:
Fixed typos and added extra clarity

This is not a finished change, as the only changes made were at the start of the page. Will finish soon

coreypjt committed:
Finish off the necessary changes

- Added new tips, warnings and information to allow users to understand any extra steps they need to take and anything they else need to know
- Added new section for downloading the SlimeVR Feeder App, which is necessary for VR tracking
- Changed various sentences and words to make instructions clearer
- Changed the Legacy Setup to go under a collapsed section as it takes a lot of space, and probably isn't going to be needed
- Fixed various typos

Let me know if I've missed something, or got something wrong :)

fineminekine commented:
> when you say SteamVR's Desktop View doesn't work whatsoever, do you mean it just doesn't load at all when you launch the flatpak app? here is the command I use to get it to run and work: flatpak run --env=WEBKIT_DISABLE_DMABUF_RENDERER=1 dev.slimevr.SlimeVR

This happened on the native app on multiple OSes. I'm currently using Kubuntu 25.10 on my PC with a dual monitor setup with Steam downloaded through the .deb package, and it still doesn't work. If there's a way to fix it, that's cool, but again, I wanted to give others a slightly easier way of fixing the desktop view. Plus, I wrote it down as an extra tip either way.

ImUrX commented on src/tools/linux-installation.md:
```diff
@@ -169,10 +198,20 @@ Example of the final directory structure:
 Once everything is all set up, all you need to do to run it is execute the AppImage and it should run everything else on its own.
 
-Note: Only tested on Debian and Ubuntu, if you use Arch please ping lordbagel42 in the SlimeVR Discord server.
+> [!WARNING]
+> It is highly recommended to open the SlimeVR server **after** you open SteamVR, as doing the opposite is known for causing desync issues with the headset, and can easily cause motion sickness, even to experienced VR users.
+
+> [!TIP]
+> If you are in VR, it is recommended to download WlxOverlay-S, which is an overlay for SteamVR which allows you to view your desktop within VR. This is recommended to download for the mounting calibration, as the positions that are requested may be hard to do without being able to follow the instructions in headset. It is also recommended to download for general use, as the default desktop shari ...
+
+> [!NOTE]
+> This was only tested on Debian and Ubuntu. If you use Arch, please notify @lordbagel42 in the [SlimeVR Discord server](https://discord.gg/slimevr).
```
its normal for it getting lot of comments dont worry, i have seen 120 comments ones :pensive:

you should probably remove the note as arch is pretty much tested with the AUR package we have

the tip i honestly think its not needed, if u want to recommend people stuff that is non-slimevr related, it will be better to provide them with a link to a better documented site

the warning is fine tho :3

fineminekine replied:
Awesome, will fix now. Thank you so much for your help! It means a lot, you have no idea, haha

coreypjt committed:
Fixed previous issues and added FAQ

- Updated dates
- Added a new Troubleshooting/FAQ section which should help new people out
- Removed a bunch of admonishes as they weren't needed
- Added a new Additional Resources section which links to the Linux VR Adventures Wiki which has a lot of tools and guides to assist in VR

kruemmelbande commented on src/tools/linux-installation.md:
```diff
@@ -121,11 +144,50 @@ You will need to open ports on:
+
+A quick and easy way to uninstall the Flatpak version is to run `flatpak uninstall dev.slimevr.SlimeVR`, which should remove the Flatpak installation of SlimeVR entirely. 
+
+After doing this, make sure you read the "Running SlimeVR" step in this guide.
```
I think there are workarounds for the flatpak to allow it access to some directory which can make it work. Im not sure of the specifics but if you can figure it out it might be worth mentioning.

fineminekine replied:
Yeah, I agree with you on that. The only thing is that, again, the guide was made for the AppImage. I might add a note beside it that identifies that it could work, but Flatpaks are annoying af to configure and I don't really have the time to fix it.

fineminekine commented on src/tools/linux-installation.md:
```diff
@@ -4,21 +4,23 @@ In order to SlimeVR to communicate with SteamVR, you'll need to install an OpenV
 ### 3. Extract and install
 
-Extract the archive you downloaded in step 1. This should give you a bunch of files and folders nested in a root `slimevr` folder. Simply move the `slimevr` folder into the `drivers` directory identified in step 2, and the plugin should now be installed. If done correctly, you should now have a `steamapps/common/SteamVR/drivers/slimevr/bin/linux64/` folder (among other things).
+Extract the archive you downloaded in step 1. This should give you a bunch of files and folders nested in a root `slimevr` folder. Simply move the `slimevr` folder into `~/.steam/steam/steamapps/common/SteamVR/drivers/`, and the plugin should now be installed. If done correctly, you should now have a `steamapps/common/SteamVR/drivers/slimevr/bin/linux64/` folder (among other things).
```
I mainly did this because I was always taught that we shouldn't refer to previous steps, and it's better to be more concise than simplistic, as it might confuse the reader

HannahPadd replied:
Installing the driver like this is not correct. User should extract driver and register it using vrpathreg.sh.

fineminekine commented on src/tools/linux-installation.md:
```diff
@@ -97,15 +99,36 @@ Parent Directory
 # Running SlimeVR
 
+> [!WARNING]
+> The Linux download located at [slimevr.dev/download](https://slimevr.dev/download) currently links to the Flatpak version of the SlimeVR Server. For this guide, it is not used, so it is recommended to download the AppImage linked below instead.
```
I mean, yeah, but the guide was explicitly made for the AppImage. I made this PR to fix some information that was incorrect or never stated prior, not for showing how to do use the other packages (like .deb and .rpm ones)

fineminekine commented on src/tools/linux-installation.md:
```diff
@@ -97,15 +99,36 @@ Parent Directory
+
+If you're receiving a warning during the automatic mounting phase mentioning that you aren't in VR mode even when you are, it's most likely due to the SlimeVR Feeder App not being installed.
+
+The AppImage release of the SlimeVR server does not include the SlimeVR Feeder App, which allows the SlimeVR Server to read the position of everything in an OpenVR instance (in this case, SteamVR) so the positions of the controllers and HMD are able to be marked as trackers. This helps with quickly and accurately seeing the rig of your full body in a small preview, and allows for tools like Virtu ...
```
Huh, didn't know that. Although, I do think that the one for Windows does. Can't test it though, because f*ck Windows.

But either way, this guide was made for the AppImage release, so that's why I said it

Smeltie commented:
Will be closing this ticket as most of these things become redundant/obsolete with the upcoming  Steam release that implements loads of fixes that also come to main distro releases of the app.

SpiritualRin commented:
And will it be possible to install it on quest 2
because it can't be installed right now

Smeltie commented:
> And will it be possible to install it on quest 2 because it can't be installed right now

There are already ways of installing it onto a quest 2, primarily via side quest.
More options are being explored, but this PR is for Desktop linux documentation, not for quest.
Furthermore this PR is closed due to the Linux version of our software getting a major overhaul making this change obsolete.
